src/lib.rs

```diff
@@ -332,179 +399,136 @@ mod getifaddrs_windows {
                                             } else {
-                                                continue 'prefixloop;
+                                                continue 'prefixloopv6;
                                             }
                                         }
                                     }
-                                    item.netmask = IpAddr::V6(Ipv6Addr::new(netmask[0],
-                                                                            netmask[1],
-                                                                            netmask[2],
-                                                                            netmask[3],
-                                                                            netmask[4],
-                                                                            netmask[5],
-                                                                            netmask[6],
-                                                                            netmask[7]));
-                                    break 'prefixloop;
+                                    item_netmask = Ipv6Addr::new(netmask[0],
+                                                                 netmask[1],
+                                                                 netmask[2],
+                                                                 netmask[3],
+                                                                 netmask[4],
+                                                                 netmask[5],
+                                                                 netmask[6],
+                                                                 netmask[7]);
+                                    break 'prefixloopv6;
                                 }
-                            }
-                        };
-                    }
-                }
-                ret.push(item);
+                                _ => continue,
+                            };
+                        }
+                        IfAddr::V6(Ifv6Addr {
+                            addr: ipv6_addr,
+                            netmask: item_netmask,
+                            broadcast: None,
+                        })
+                    },
+                };
+                ret.push(Interface {
+                    name: name,
+                    addr: addr,
+                });
             }
         }
         unsafe {
             libc::free(ifaddrs as *mut c_void);
         }
-        ret
+        Ok(ret)
     }
 }
 #[cfg(windows)]
 /// Get address
-pub fn getifaddrs() -> Vec<IfAddr> {
-    getifaddrs_windows::getifaddrs()
-}
-
-fn is_loopback(ip_addr: &IpAddr) -> bool {
-    match *ip_addr {
-        IpAddr::V4(a) => a.octets()[0] == 127,
-        IpAddr::V6(a) => a.segments() == [0, 0, 0, 0, 0, 0, 0, 1],
-    }
-}
-
-/// Remove loopback address(s)
-pub fn filter_loopback(mut ifaddrs: Vec<IfAddr>) -> Vec<IfAddr> {
```

Comment:
That's right. I added an `is_loopback` method to the `Interface`, `IfAddr`, `Ifv4Addr` and `Ifv6Addr` types. If someone's iterating over interfaces and they want to filter out the loopback interfaces they can call `.filter(|i| !i.is_loopback())`. This seems a bit clearer and a bit more normal to me.